src/hooks/usePublisher.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect, useRef, useState } from "react";
+import { Director, Publish } from '@millicast/sdk';
+import type { Publisher, PublisherState } from './types/Publisher';
+
+
+const usePublisher = (token: string, streamName: string): Publisher => {
+
+    const [publisherState, setPublisherState] = useState<PublisherState>("ready");
+    const publisher = useRef<Publish>();
+
+    useEffect(() => {
+        const tokenGenerator = () => Director.getPublisher({ token: token, streamName: streamName });
+        publisher.current = new Publish(streamName, tokenGenerator);
+    }, [token, streamName]);
+
+    // TODO, this param list can grow significantly when we add the broadcast settings option, but until such time this list will stay small
+    const startStreaming = async (mediaStream: MediaStream) => {
+
+        if (!publisher.current || publisher.current.isActive() || publisherState !== "ready") return;
+        const broadcastOptions = {
+            mediaStream,
+            // Additional settings go here
+        };
+        try {
+            console.log(`Connecting with ${token} and ${streamName}`);
+            setPublisherState("connecting");
+            await publisher.current.connect(broadcastOptions);
+            setPublisherState("streaming")
+        } catch (e) {
+            setPublisherState("ready");
+            console.error(e);
+        }
+    };
+
+    const stopStreaming = async () => {
+        if (!publisher.current || publisher.current.isActive() || publisherState !== "streaming") return;
```

Comment:
Is that OK to return when it's `connecting`?  I think it could be publisher.current?.stop() which is way easier.